### PR TITLE
Fixed issues with Error responses / RestException

### PR DESCRIPTION
**Pull request by jackjwilliams**
RealWorld errors should be in the form of:

```
{
  "errors":{
    "body": [
      "can't be empty"
    ]
  }
}
```

According to https://github.com/gothinkster/realworld/tree/master/api#errors-and-status-codes

There were issues when registing a new user if the email / username already existed (it didn't display any errors, because we only returned HttpStatusCode.BadRequest).

Fleshed out / cleaned up the ErrorHandlingMiddleware a little, added appropriate error messages to instances of RestExceptions. Now errors should display properly in most front ends.

### CODE REVIEW

**Comment by jackjwilliams** [comment]
Not quite done yet, still a few more changes to come.

**Comment by jackjwilliams** [comment]
Hrmm, I rebased on master to get the latest ValidationPipelineBehavior changes, now anytime I login then try to view the homepage I get the following exception:

Unable to resolve service for type 'FluentValidation.IValidator`1[Conduit.Features.Tags.List+Query]' while attempting to activate Conduit.Infrastructure.ValidationPipelineBehavior`2[Conduit.Features.Tags.List+Query,Conduit.Features.Tags.TagsEnvelope]'.

File: src/Conduit/Infrastructure/Security/JwtIssuerOptions.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace Conduit.Infrastructure.Security
 {
     public class JwtIssuerOptions
     {
-        public const string Scheme = "Token";
+        public const string Scheme = "Bearer";
```
Comment (jackjwilliams) [inline src/Conduit/Infrastructure/Security/JwtIssuerOptions.cs]: This was an accident and specific to my project, I will remove.

**Comment by VictorioBerra** [comment]
It might be nice to tell Swagger what our 400 will look like. Maybe consider using [this approach or package?](https://github.com/ASP-NET-Core-Boilerplate/Framework/blob/master/Source/Boilerplate.AspNetCore.Swagger/SchemaFilters/ModelStateDictionarySchemaFilter.cs).

Also, the ModelStateDictionary returns something in a very similar structure to this. Not sure if we could just return that, or wrap and return it with an ActionFilter?

**Comment by adamhathcock** [comment]
This looks good to me.  Sorry I left it so long.  Fixing merge issues

**Pull request merged**